pkg/resource/deploy/plan_executor.go

```diff
@@ -153,11 +154,14 @@ func (pe *planExecutor) Execute(callerCtx context.Context, opts Options, preview
 					return false, nil
 				}
 
-				if eventErr := pe.handleSingleEvent(event.Event); eventErr != nil {
-					log.Infof("planExecutor.Execute(...): error handling event: %v", eventErr)
-					pe.reportError(pe.plan.generateEventURN(event.Event), eventErr)
+				if res := pe.handleSingleEvent(event.Event); res != nil {
+					resErr := res.Error()
+					if resErr != nil {
+						log.Infof("planExecutor.Execute(...): error handling event: %v", resErr)
+						pe.reportError(pe.plan.generateEventURN(event.Event), resErr)
+					}
 					cancel()
-					return false, eventErr
+					return false, result.TODO()
```

Comment:
Not sure if this is reasonable API design, but part of me wants to add a string (literal) argument to `result.TODO` that forces users to write a comment as to why this TODO is necessary. It could be an actual code comment here, todo.